core/drivers/firewall/stm32_risaf.c

```diff
@@ -0,0 +1,770 @@
+		    struct stm32_risaf_region *region, unsigned int index)
+{
+	unsigned int i = 0;
+
+	for (i = 0; i < index; i++) {
+		/* Skip region if there's no configuration */
+		if (!region[i].cfg)
+			continue;
+
+		if (core_is_buffer_intersect(region[index].addr,
+					     region[index].len,
+					     region[i].addr,
+					     region[i].len)) {
+			EMSG("RISAF %#"PRIxPTR": Regions %u and %u overlap",
+			     risaf->pdata.base.pa, index, i);
+			return TEE_ERROR_GENERIC;
+		}
+	}
+
+	return TEE_SUCCESS;
+}
+
+static TEE_Result risaf_configure_region(struct stm32_risaf_instance *risaf,
+					 uint32_t region_id, uint32_t cfg,
+					 uint32_t cid_cfg, uintptr_t saddr,
+					 uintptr_t eaddr)
+{
+	uint32_t mask = risaf->ddata->mask_regions;
+	vaddr_t base = risaf_base(risaf);
+
+	io_clrbits32(base + _RISAF_REG_CFGR(region_id), _RISAF_REG_CFGR_BREN);
+
+	io_clrsetbits32(base + _RISAF_REG_STARTR(region_id), mask,
+			(saddr - risaf->pdata.mem_base) & mask);
+	io_clrsetbits32(base + _RISAF_REG_ENDR(region_id), mask,
+			(eaddr - risaf->pdata.mem_base) & mask);
+	io_clrsetbits32(base + _RISAF_REG_CIDCFGR(region_id),
+			_RISAF_REG_CIDCFGR_ALL_MASK,
+			cid_cfg & _RISAF_REG_CIDCFGR_ALL_MASK);
+
+	io_clrsetbits32(base + _RISAF_REG_CFGR(region_id),
+			_RISAF_REG_CFGR_ALL_MASK,
+			cfg & _RISAF_REG_CFGR_ALL_MASK);
+
+	if (cfg & _RISAF_REG_CFGR_ENC) {
+		if (!risaf->pdata.enc_supported) {
+			EMSG("RISAF %#"PRIxPTR": encryption feature error",
+			     risaf->pdata.base.pa);
+			return TEE_ERROR_GENERIC;
+		}
+
+		if ((cfg & _RISAF_REG_CFGR_SEC) != _RISAF_REG_CFGR_SEC) {
+			EMSG("RISAF %#"PRIxPTR": encryption on non secure area",
+			     risaf->pdata.base.pa);
+			return TEE_ERROR_GENERIC;
+		}
+	}
+
+	DMSG("RISAF %#"PRIxPTR": region %02d - start 0x%08x - end 0x%08x - cfg 0x%08x - cidcfg 0x%08x",
+	     risaf->pdata.base.pa, region_id,
+	     io_read32(base + _RISAF_REG_STARTR(region_id)),
+	     io_read32(base + _RISAF_REG_ENDR(region_id)),
+	     io_read32(base + _RISAF_REG_CFGR(region_id)),
+	     io_read32(base + _RISAF_REG_CIDCFGR(region_id)));
+
+	return TEE_SUCCESS;
+}
+
+static void risaf_print_version(struct stm32_risaf_instance *risaf)
+{
+	vaddr_t base = risaf_base(risaf);
+	struct stm32_risaf_version __maybe_unused version = {
+		.major = (io_read32(base + _RISAF_VERR) &
+			  _RISAF_VERR_MAJREV_MASK) >> _RISAF_VERR_MAJREV_SHIFT,
+		.minor = (io_read32(base + _RISAF_VERR) &
+			  _RISAF_VERR_MINREV_MASK) >> _RISAF_VERR_MINREV_SHIFT,
+		.ip_id = io_read32(base + _RISAF_IPIDR),
+		.size_id = io_read32(base + _RISAF_SIDR)
+	};
+
+	DMSG("RISAF %#"PRIxPTR" version %d.%d, ip0x%x size0x%x",
+	     risaf->pdata.base.pa, version.major, version.minor, version.ip_id,
+	     version.size_id);
+}
+
+/*
+ * @brief  Lock the RISAF IP registers for a given instance.
+ * @param  risaf: RISAF instance.
+ * @retval 0 if OK, negative value else.
+ */
+static __maybe_unused
+int stm32_risaf_lock(struct stm32_risaf_instance *risaf)
```

Comment:
It would be better to return a `TEE_Result` code, or even better `void` if intended to always succeed.